zenoh-ts/src/session.ts

```diff
@@ -0,0 +1,624 @@
+*/
+export interface DeleteOpts {
+  congestion_control?: CongestionControl,
+  priority?: Priority,
+  express?: boolean,
+  attachment?: IntoZBytes
+}
+
+/**
+ * Options for a Get function 
+ * @prop {ConsolidationMode=} consolidation - consolidation mode
+ * @prop {CongestionControl=} congestion_control - congestion_control applied when routing the data
+ * @prop {Priority=} priority - priority of the written data
+ * @prop {boolean=} express  - Express 
+ * @prop {Encoding=} encoding - Encoding type of payload 
+ * @prop {IntoZBytes=} payload - Payload associated with getrequest
+ * @prop {IntoZBytes=} attachment - Additional Data sent with the request
+*/
+export interface GetOptions {
+  consolidation?: ConsolidationMode,
+  congestion_control?: CongestionControl,
+  priority?: Priority,
+  express?: boolean,
+  encoding?: Encoding,
+  payload?: IntoZBytes,
+  attachment?: IntoZBytes
+}
+
+/**
+ * Options for a Queryable
+ * @prop complete - Change queryable completeness.
+ * @prop callback - Callback function for this queryable
+*/
+export interface QueryableOpts {
+  complete?: boolean,
+  callback?: (query: Query) => void,
+}
+
+/**
+ *  Set of options used when declaring a publisher
+ * @prop {Encoding} encoding - Optional, Type of Encoding data to be sent over
+ * @prop {CongestionControl} congestion_control - Optional, Type of Congestion control to be used (BLOCK / DROP)
+ * @prop {Priority} priority - Optional, The Priority of zenoh messages
+ * @prop {boolean} express - Optional, The Priority of zenoh messages
+ * @prop {Reliability} reliability - Optional, The Priority of zenoh messages
+ */
+export interface PublisherOptions {
+  encoding?: Encoding,
+  congestion_control?: CongestionControl,
+  priority?: Priority,
+  express?: boolean,
+  reliability?: Reliability,
+}
+
+// ███████ ███████ ███████ ███████ ██  ██████  ███    ██
+// ██      ██      ██      ██      ██ ██    ██ ████   ██
+// ███████ █████   ███████ ███████ ██ ██    ██ ██ ██  ██
+//      ██ ██           ██      ██ ██ ██    ██ ██  ██ ██
+// ███████ ███████ ███████ ███████ ██  ██████  ██   ████
+
+/**
+ * Zenoh Session
+ */
+export class Session {
+  // WebSocket Backend
+  private remote_session: RemoteSession;
+  /** Finalization registry used for cleanup on drop
+   * @hidden 
+   */
+  static registry: FinalizationRegistry<RemoteSession> = new FinalizationRegistry((r_session: RemoteSession) => r_session.close());
+
+  dispose() {
+    this.close();
+    Session.registry.unregister(this);
+  }
+
+  private constructor(remote_session: RemoteSession) {
+    this.remote_session = remote_session;
+    Session.registry.register(this, remote_session, this)
+  }
+
+  /**
+   * Creates a new Session instance
+   *
+   * @remarks
+   *  Opens A Zenoh Session
+   *
+   * @param config - Config for session
+   * @returns Typescript instance of a Session
+   *
+   */
+
+  static async open(config: Promise<Config> | Config): Promise<Session> {
+    const cfg = await config;
```

Comment:
I don't think we should `await` a non-Promise, see this [StackOverflow question](https://stackoverflow.com/questions/55262996/does-awaiting-a-non-promise-have-any-detectable-effect).

Reply:
Good catch ! fixed